labcodes/labstorybook
# DSYS-23: Dialogs

**Link to task:**
https://labcodes.atlassian.net/browse/DSYS-23

**Description:**
This PR implements the JS behavior of `StandardDialog` and `MessageDialog`. It introduces a `DialogPlayground.js` for the documentation on `Dialog.stories.mdx`. It also introduces the backbone for styling in `_lab-dialog.scss`. 

A new dependency `react-focus-trap` is added to the project for trapping the focus inside modal _Dialogs_. Therefore, make sure to run `npm install` inside the directory `labstorybook/labsystem/`.

**How to test:**
- Check if the `Dialog` tab appears at the left sidebar.
- Check if `Playground` session is visible and works well with varied input. Edit `Configurations` at the right side, then press `Show Modal` button.
--
- With `isModal` **checked**, press `Show Modal`, then try to move focus around with TAB. Check if focus is **trapped** and scroll down is **locked**;
- With `isModal` **unchecked**, press `Show Modal`, then try to move focus around with TAB. Check if focus is **not trapped** and scroll down is **unlocked**;
- With `isModal` **unchecked**, press `Show Modal`, then reduce the window size until reaching mobile width. Check if overlay **appears**, focus is **trapped**, and scroll down is **locked**. Now increase the window size until crossing that threshold. Check if overlay **disappears**, focus is **released** and scroll down is **unlocked**.
- When _overlay_ is **on**, check if _Dialog_ **closes** when clicking outside of _Dialog_;
- When _overlay_ is **off**, check if _Dialog_ **does not close** when clicking outside of _Dialog_;
- When _Dialog_ is **open**, close by clicking on `Close button` (X);
- When _Dialog_ is **open**, close by clicking on `Swipe down button` (v);
--
- With mobile emulator **enabled** on dev tools, click and hold on `Swipe down button`, swipe at least 75px to the bottom, then release. Check if _Dialog_ **closes**.
